src/Xamarin.Android.Build.Tasks/Tasks/AsyncTask.cs

```diff
@@ -12,9 +12,11 @@ public class AsyncTask : Task, ICancelableTask {
 		Queue logMessageQueue =new Queue ();
 		Queue warningMessageQueue = new Queue ();
 		Queue errorMessageQueue = new Queue ();
+		Queue customMessageQueue = new Queue ();
```

Comment:
Shouldn't we be using [`Queue<T>`](https://msdn.microsoft.com/en-us/library/7977ey2c(v=vs.110).aspx)?

Reply:
`Queue<T>` does not include a `SyncRoot` property which we need for threading. Now I can add a bunch of extra objects to handle that. Remember this code was sourced from the msbuild ToolTask. At the time it was safer to follow their lead rather than introduce possible unknowns.